server/src/immich/api-v1/asset/asset.service.ts

```diff
@@ -566,66 +532,15 @@ export class AssetService {
     return { filepath: asset.resizePath, contentType: 'image/jpeg' };
   }
 
-  private async streamFile(filepath: string, res: Res, headers: Record<string, string>, contentType?: string | null) {
-    await fs.access(filepath, constants.R_OK);
-    const { size, mtimeNs } = await fs.stat(filepath, { bigint: true });
-
+  async streamFile(res: Res, filepath: string, contentType?: string | null) {
     if (contentType) {
-      res.header('Content-Type', contentType);
+      res.contentType(contentType);
     }
-
-    const range = this.setResRange(res, headers, Number(size));
-
-    // etag
-    const etag = `W/"${size}-${mtimeNs}"`;
-    res.setHeader('ETag', etag);
-    if (etag === headers['if-none-match']) {
-      res.status(304);
-      return;
-    }
-
-    const stream = createReadStream(filepath, range);
-    return await pipeline(stream, res).catch((err) => {
-      if (err.code !== 'ERR_STREAM_PREMATURE_CLOSE') {
+    const options = filepath.startsWith('/') ? {} : { root: process.cwd() }; // handle relative paths
```

Comment:
why cwd? Isn't the actual path set elsewhere?

Reply:
For whatever reason the url in the database is a relative path. The cwd for the app is `/usr/src/app`, which is set in the `Dockerfile`.

Reply:
Relative to what though? We also can't ever be certain the path will be /usr/src/app.

Reply:
Shouldn't it be relative to the UPLOAD_LOCATION env var?

Reply:
`UPLOAD_LOCATION` is for the path outside of the container. Internally it's always `/usr/src/app/upload`. The relative paths, working directories and mounts all assume `/usr/src/app/upload`. This could probably be made into an env variable, but that's for another PR.